org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/code/makestatic/ContextCalculator.java

```diff
@@ -0,0 +1,253 @@
+		fSelectionASTNode= NodeFinder.perform(fSelectionCompilationUnit, fSelectionEditorText.getOffset(), fSelectionEditorText.getLength());
+
+		if (fSelectionASTNode == null) {
+			return;
+		}
+
+		if (fSelectionASTNode.getNodeType() == ASTNode.SIMPLE_NAME) {
+			fSelectionASTNode= fSelectionASTNode.getParent();
+		} else if (fSelectionASTNode.getNodeType() == ASTNode.EXPRESSION_STATEMENT) {
+			fSelectionASTNode= ((ExpressionStatement) fSelectionASTNode).getExpression();
+		}
+	}
+
+	/**
+	 * This method calculates the {@link #fTargetIMethodBinding}. If the {@link #fSelectionASTNode}
+	 * is an instance of {@link MethodInvocation} or {@link MethodDeclaration}, it resolves the
+	 * corresponding {@link IMethodBinding} from that instance.
+	 */
+	private void calculateTargetIMethodBinding() {
+		if (getOrComputeSelectionASTNode() == null) {
+			return;
+		}
+
+		if (fSelectionASTNode instanceof MethodInvocation selectionMethodInvocation) {
+			fTargetIMethodBinding= selectionMethodInvocation.resolveMethodBinding();
+		} else if (fSelectionASTNode instanceof MethodDeclaration selectionMethodDeclaration) {
+			fTargetIMethodBinding= selectionMethodDeclaration.resolveBinding();
+		}
+	}
+
+	/**
+	 * Calculates the target IMethod from the target IMethodBinding. This method retrieves the
+	 * IMethod represented by the {@link #fTargetIMethodBinding} and assigns it to
+	 * {@link #fTargetIMethod}
+	 */
+	private void calculateTargetIMethod() {
+		if (getOrComputeTargetIMethodBinding() == null) {
+			return;
+		}
+
+		fTargetIMethod= (IMethod) fTargetIMethodBinding.getJavaElement();
+	}
+
+	/**
+	 * Calculates the target ICompilationUnit from the target IMethod. This method retrieves the
+	 * declaring type of the {@link #fTargetIMethod}, then gets its associated compilation unit, and
+	 * assigns it to {@link #fTargetICompilationUnit}.
+	 */
+	private void calculateTargetICompilationUnit() {
+		if (getOrComputeTargetIMethod() == null) {
+			return;
+		}
+		fTargetICompilationUnit= fTargetIMethod.getDeclaringType().getCompilationUnit();
+	}
+
+	/**
+	 * Converts the target {@link ICompilationUnit} to a {@link CompilationUnit} and assigns it to
+	 * {@link #fTargetCompilationUnit}.
+	 */
+	private void calculateTargetCompilationUnit() {
+		if (getOrComputeTargetICompilationUnit() == null) {
+			return;
+		}
+		fTargetCompilationUnit= convertICompilationUnitToCompilationUnit(fTargetICompilationUnit);
+	}
+
+	/**
+	 * Resolves the method declaration and binding for the target method.
+	 *
+	 * @throws JavaModelException if the target method or the target compilationUnit is invalid.
+	 */
+	private void calculateMethodDeclaration() throws JavaModelException {
+		calculateTargetCompilationUnit();
+		if (fTargetCompilationUnit == null) {
+			return;
+		}
+		fTargetMethodDeclaration= getMethodDeclarationFromIMethod(fTargetIMethod, fTargetCompilationUnit);
+		fTargetIMethodBinding= fTargetMethodDeclaration.resolveBinding();
```

Comment:
You have to dive deeo into the code to see that `fTargetIMethod` will never be `null` because of calling `calculateTargetCompilationUnit()` before. It would be even better if all the `calculate...` methods first call the `getOrCompute...` methods for the variables they require and store the values in local variables that they access later on. Then you can check all the values for not being `null` and safely use them afterwards. Then it is (and will always be even upon changes to the class) guaranteed that those values are never `null` when using them.

Here, e.g., it would look like this:
```suggestion
		IMethod targetIMethod = getOrComputeTargetIMethod();
		CompilationUnit targetCompilationUnit = getOrComputeTargetCompilationUnit();
		if (targetIMethod == null || targetCompilationUnit == null) {
			return;
		}
		fTargetMethodDeclaration= getMethodDeclarationFromIMethod(targetIMethod, targetCompilationUnit);
		fTargetIMethodBinding= fTargetMethodDeclaration.resolveBinding();
```

Reply:
Done.
2a75304fb32d8daeccb0ad773ace6e755e9b6e43